src/integration/java/io/pinecone/DataPlane/FutureStubTest.java

```diff
@@ -0,0 +1,215 @@
+package io.pinecone.DataPlane;
+
+import com.google.common.primitives.Floats;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.protobuf.Struct;
+import com.google.protobuf.Value;
+import io.pinecone.*;
+import io.pinecone.helpers.RandomStringBuilder;
+import io.pinecone.model.IndexMeta;
+import io.pinecone.proto.*;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class FutureStubTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(FutureStubTest.class);
+    public String indexName = "integ-test-sanity";
+
+    private PineconeClient dataPlaneClient;
+    private PineconeIndexOperationClient controlPlaneClient;
+
+    @BeforeEach
+    public void setUp() {
+        PineconeClientConfig config = new PineconeClientConfig()
+                .withApiKey(System.getenv("PINECONE_API_KEY"))
+                .withEnvironment(System.getenv("PINECONE_ENVIRONMENT"))
+                .withServerSideTimeoutSec(10);
+
+        controlPlaneClient = new PineconeIndexOperationClient(config);
+        dataPlaneClient = new PineconeClient(config);
+    }
+
+    @Test
+    public void checkIndexSetup() throws Exception {
+        IndexMeta indexMeta = controlPlaneClient.describeIndex(indexName);
+        assertEquals(3, indexMeta.getDatabase().getDimension());
+    }
+
+    @Test
+    public void sanity() throws Exception {
```

Comment:
I think this test is doing way too much. When a test like this fails, it takes a lot of investigation to figure out what is actually breaking.

Some operations (e.g. create & describe/list, upsert & fetch, upsert & query) make sense to use and test together, but you could still break this down into many smaller test cases.

I would have test cases for:
- Create index (with minimal params) and describe index
- Create index (passing all optional params) and describe index
- Upsert and fetch
- Upsert hybrid and fetch 
- Upsert and fetch by id
- Upsert and query by id
- Upsert and query with vector values
  - with `includeMetadata`
  - with `includeValues` 
- Upsert hybrid and query
- Edge cases (missing params, invalid names, wrong vector dimension, etc)

Reply:
This is great, thank you for providing me with this list. It has been like this since I picked up and always thought it was okay haha.